src/main/java/agh/ics/oop/Animal.java

```diff
@@ -18,6 +22,7 @@ public Animal(IWorldMap map, Vector2d initialPosition) {
         this.orientation = MapDirection.NORTH;
         this.position = initialPosition;
         this.map = map;
+        this.addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.